packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/models/TypeVariableModel.java

```diff
@@ -43,7 +64,8 @@ public Class<TypeVariableModel> getCommonModelClass() {
 
     @Override
     public int hashCode() {
-        return Objects.hash(getName(), resolve().hashCode());
+        return Objects.hash(getCommonModelClass().getName(), getName(),
+                getAnnotations(), resolve());
```

Comment:
```suggestion
        return 0x4f76c9f1 ^ getName().hashCode();
```
This looks better to me performance-wise.